File: java/board/service/article/src/main/java/vedivero/board/article/controller/ArticleController.java
package vedivero.board.article.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import vedivero.board.article.service.ArticleService;
import vedivero.board.article.service.request.ArticleCreateRequest;
import vedivero.board.article.service.response.ArticlePageResponse;
import vedivero.board.article.service.response.ArticleResponse;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class ArticleController {

    private final ArticleService articleService;

    @GetMapping("/v1/articles/{articleId}")
    public ArticleResponse read(@PathVariable Long articleId) {
        return articleService.read(articleId);
    }

    @GetMapping("/v1/articles")
    public ArticlePageResponse readAll(
            @RequestParam("boardId") Long boardId, // 게시판 번호
            @RequestParam("page") Long page, //조회할 페이지 번호
            @RequestParam("pageSize") Long pageSize // 한 페이지에 표시할 컨텐츠 수
    ){
        return articleService.readAll(boardId, page, pageSize);
    }

    @GetMapping("/v1/articles/infinite-scroll")
    public List<ArticleResponse> readAllInfiniteScroll(
            @RequestParam("boardId") Long boardId, // 게시판 번호
            @RequestParam("pageSize") Long pageSize, // 조회할 컨텐츠 개수
            @RequestParam(value = "lastArticleId", required = false) Long lastArticleId //
            ) {
        return articleService.readAllInfiniteScroll(boardId, pageSize, lastArticleId);
    }

    @PostMapping("/v1/articles")
    public ArticleResponse create(@RequestBody ArticleCreateRequest request) {
        return articleService.create(request);
    }

    @PutMapping("/v1/articles/{articleId}")
    public ArticleResponse update(@PathVariable Long articleId, @RequestBody ArticleCreateRequest request) {
        return articleService.update(articleId, request);
    }

    @DeleteMapping("/v1/articles/{articleId}")
    public void delete(@PathVariable Long articleId) {
        articleService.delete(articleId);
    }
}

File: kotlin/three-dimensional-globe/src/main/java/com/globe/three_dimensional_globe/controller/ArticleController.kt
package com.globe.three_dimensional_globe.controller

import com.globe.three_dimensional_globe.dto.ArticleRequest
import com.globe.three_dimensional_globe.dto.ArticleResponse
import com.globe.three_dimensional_globe.service.ArticleService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/article")
class ArticleController(
    private val articleService: ArticleService
) {

    @PostMapping
    fun createArticle(
        @RequestBody articleRequest:ArticleRequest
    ): ArticleResponse{
        return articleService.createArticle(articleRequest)
    }

    @GetMapping
    fun findAll(): List<ArticleResponse> {
        return articleService.findAll()
    }

}